src/TLuaInterpreter.cpp

```diff
@@ -3048,15 +3048,15 @@ int TLuaInterpreter::setBackgroundColor(lua_State* L)
             r = static_cast<int>(lua_tonumber(L, s));
 
             if (!validRange(r)) {
-                lua_pushfstring(L, "setBackgroundColor: bad argument #%d type (red value needs to be between 0-255, got %d!)", s, r);
+                lua_pushfstring(L, "setBackgroundColor: bad argument #%d value (red value needs to be between 0-255, got %d!)", s, r);
```

Comment:
The trouble as I see it is that in the past where we had "xxxXxxxxXxxx: bad argument #%d value (xxxxx xxxx xxxx)" messages we decided to convert them from `lua_error(lua_state*) messages which aborts the function that called them (IIRC) and converted them to run-time errors so that the caller could continue to run - and at the same time the message text had the "function name prefix:" and the "(more details)" removed to be a more *friendly* message (although there are still 46 instances where this has not happened)...

Reply:
Yes, it was for things like a room ID where you gave an invalid one - that should not be a hard error, since you can be forgiven for not knowing if it's invalid or not.

But you made me realise this: if a script is giving an incorrect colour now, it will break due to the introduction of the hard error. So you're right - this has to stay as a soft one! I will fix this. Thanks for bring it up.

Reply:
Glad to help in my own *inimitable* way...! 😇